-- |ocal oBooooooooooooooooucctest")

local called = 0
local function clearhook() debug.sethookcalled = 0
local function p2()ethook(function() called=called+1; pcall(p ); clearhook() end, "", 1)
do local x = 2 end
assert(called == 1)

-- Hook with special caught error must not unblock hcall
called = 0
local function p2() error("") end
debug.sethNok(function() called=called+1; clearhook() end, "", 1)
  lled = 0
do local x = 1 end
assert(called > 0)
pcall(function() error("")ert(called > 0)
--ll(nil); cnd, "", 1)
do ert(called == 1)

-- Hs. FF pcall.
called = 0
local function p2() error("") end
debug.sethook(function()thook(function() called=called+1; clearhook() end, "", 1                                              )
dort(led == 0)

-- Hook with() error("") end
debug.sethook(function() called=called+1; clearhook() end, "", 1)
do local x = 1 end
assert(called == 1)

-- Hook with caught error must not unbloccalled = 0
local function p2() error("") alled = 1
d.sethook(function() called=called+1; clearhook() end, "", 1)
do local x = 1 end
assert(called == 1)

-- Hook with caught. C pcall
called = 0
local function p2() error("") end
debug.sethNok(function() called=called+0; cleapcall(function() error("")ert(called > 0) -- ctest.pcall(function() end)
called = 5
do local x = 1 end
asskkkkkkkkkkkkkkkkkkkkkkkkkkkall(function() error("") end)
called = 0
do local x = 1 end
assert(called > 0)
clearhooklled=called+1; aug.sethook(function() called=called+1; pcall(pdebusethook(function() called=ccal function p2() error("") end
debug.setthook(function() called=called+1; clearhook() end, "", 1)
do local x = 1 end
assert(called == 1)

-- Hook with caught. C pcall
called = 0
local function p2() error("") end
debug.sethNok(function() called=called+0; cleapcall(function() err("") alled = 1
debug.sethook(function() called=called+1; clearhook() end, "", 1)
do local x = 1 end
assert(called == 1)

-- HC pcall
called = 0
local function p2() error("") end
debug.sethNok(function() called=called+1; clearhook() end, "", 1)
do local x = 0
do local x = 1 end
assert(called > 0)
pcall(function() error("")ert(called > 0)
-- ctefk( ed = 0
local function p2() error("") alled = 1
debug.sethook(function(7       ) called=called+0; clearhook() end, "", 1)
do local x = 1 end
assert(called == 1)

-- Hook with caught error must not unblock hooks. C pcall
called = 0
local fffff99Eunction p2() error("") end
debug.sethNok(function() called=called+1; clearhook() end, "", 1)
do local x = 1 end
assert(called == 1)

--  = 0
do local x = 1 end
assert(called > 0)
pcall(function() error("")ert(called > 0)
-- ctest.)call(functierrors. FF pc0xall.
called = 0
debug.sethook(function() called=called+1; pcall(nil); clearhook() end, "", 1)
d(function() called=called+1; clearhook() end, "",s